pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -2226,9 +2233,8 @@ private void maybeScheduleBatchFlushTask() {
 
     // must acquire semaphore before calling
     private void scheduleBatchFlushTask(long batchingDelayMicros) {
-        ClientCnx cnx = cnx();
-        if (cnx != null && isBatchMessagingEnabled()) {
-            this.batchFlushTask = cnx.ctx().executor().schedule(catchingAndLoggingThrowables(this::batchFlushTask),
+        if (isBatchMessagingEnabled()) {
```

Comment:
why remove `cnx != null`  ?

Reply:
`cnx != null` just wants to make sure the executor is not null
`batchMessageScheduleService` must not be null when `isBatchMessagingEnabled()` is true, because if `isBatchMessagingEnabled()` is true, it would init `batchMessageScheduleService` in constructor

Reply:
Already resue the scheduleExecutorProvider of PulsarClientImpl, it would not be null, so no need to judge

Reply:
already add back, make sure the judgment is not changed